core/services/pipeline/task.base.go

```diff
@@ -24,6 +24,8 @@ type BaseTask struct {
 
 	Tags string `mapstructure:"tags" json:"-"`
 
+	StreamID null.Uint32 `mapstructure:"streamID"`
```

Comment:
Is this really necessary @samsondav ? This feels like we're adding things that are very implementation specific to a general domain -- I admittedly don't understand the full context though

Reply:
It is implementation-specific unfortunately but there isn't really any other way to do it